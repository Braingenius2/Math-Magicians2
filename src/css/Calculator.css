.calculator {
  display: flex;
  flex-direction: column;
  width: 40%;
  background-color: #333;
  border-radius: 5px;
}

.display {
  background-color: #969696;
  padding: 0.8rem;
}

.display input {
  width: 100%;
  background-color: transparent;
  border: none;
  color: white;
  font-size: 24px;
  text-align: right;
}

.buttons {
  display: flex;
  flex-wrap: wrap;
}

.row {
  display: flex;
  width: 100%;
}

button {
  flex: 1;
  padding: 15px;
  font-size: 18px;
  color: black;
  border-bottom: gray;
  border-right: gray;
  cursor: pointer;
  outline: none;
}

button:hover {
  background-color: #555;
}

button.operation {
  background-color: #f5923e;
  color: white;
}

button.operation:hover {
  background-color: #f5a55a;
}

button.zero {
  flex: 2;
  padding: 15px 30px;
}

.calculator-page {
  display: flex;
  justify-content: space-between;
}
