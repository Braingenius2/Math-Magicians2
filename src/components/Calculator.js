import React, { useState } from 'react';
import Button from './Button';
import '../css/Calculator.css';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState({ total: null, next: null, operation: null });
  const handleClick = (buttonName) => {
    setState(calculate(state, buttonName));
  };

  return (
    <div className="calculator">
      <div className="display">
        <input
          type="text"
          readOnly
          value={state.next || state.total || '0'}
        />
      </div>
      <div className="buttons">
        <div className="row">
          <Button onClick={() => handleClick('AC')} label="AC" />
          <Button onClick={() => handleClick('+/-')} label="+/-" />
          <Button onClick={() => handleClick('%')} label="%" />
          <Button onClick={() => handleClick('÷')} className="operation" label="÷" />
        </div>
        <div className="row">
          <Button onClick={() => handleClick('7')} className="number" label="7" />
          <Button onClick={() => handleClick('8')} className="number" label="8" />
          <Button onClick={() => handleClick('9')} className="number" label="9" />
          <Button onClick={() => handleClick('×')} className="operation" label="×" />
        </div>
        <div className="row">
          <Button onClick={() => handleClick('4')} label="4" />
          <Button onClick={() => handleClick('5')} label="5" />
          <Button onClick={() => handleClick('6')} label="6" />
          <Button onClick={() => handleClick('-')} className="operation" label="-" />
        </div>
        <div className="row">
          <Button onClick={() => handleClick('1')} className="number" label="1" />
          <Button onClick={() => handleClick('2')} className="number" label="2" />
          <Button onClick={() => handleClick('3')} className="number" label="3" />
          <Button onClick={() => handleClick('+')} className="operation" label="+" />
        </div>
        <div className="row">
          <Button onClick={() => handleClick('0')} className="number zero" label="0" />
          <Button onClick={() => handleClick('.')} label="." />
          <Button onClick={() => handleClick('=')} className="operation" label="=" />
        </div>
      </div>
    </div>
  );
};

export default Calculator;
