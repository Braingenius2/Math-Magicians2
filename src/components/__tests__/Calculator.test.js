import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Calculator from '../Calculator';

describe('Calculator', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Calculator />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should update the display when a button is clicked', () => {
    const { getByText, getByDisplayValue } = render(<Calculator />);
    const button = getByText('7');
    fireEvent.click(button);
    const display = getByDisplayValue('7');
    expect(display).toBeInTheDocument();
  });
});
