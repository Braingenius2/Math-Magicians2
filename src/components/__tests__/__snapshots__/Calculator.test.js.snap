// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator should render correctly 1`] = `
<DocumentFragment>
  <div
    class="calculator-page"
  >
    <div>
      <h2>
        Let's do some math!
      </h2>
    </div>
    <div
      class="calculator"
    >
      <div
        class="display"
      >
        <input
          readonly=""
          type="text"
          value="0"
        />
      </div>
      <div
        class="buttons"
      >
        <div
          class="row"
        >
          <button
            class=""
            type="button"
          >
            AC
          </button>
          <button
            class=""
            type="button"
          >
            +/-
          </button>
          <button
            class=""
            type="button"
          >
            %
          </button>
          <button
            class="operation"
            type="button"
          >
            รท
          </button>
        </div>
        <div
          class="row"
        >
          <button
            class="number"
            type="button"
          >
            7
          </button>
          <button
            class="number"
            type="button"
          >
            8
          </button>
          <button
            class="number"
            type="button"
          >
            9
          </button>
          <button
            class="operation"
            type="button"
          >
            ร
          </button>
        </div>
        <div
          class="row"
        >
          <button
            class=""
            type="button"
          >
            4
          </button>
          <button
            class=""
            type="button"
          >
            5
          </button>
          <button
            class=""
            type="button"
          >
            6
          </button>
          <button
            class="operation"
            type="button"
          >
            -
          </button>
        </div>
        <div
          class="row"
        >
          <button
            class="number"
            type="button"
          >
            1
          </button>
          <button
            class="number"
            type="button"
          >
            2
          </button>
          <button
            class="number"
            type="button"
          >
            3
          </button>
          <button
            class="operation"
            type="button"
          >
            +
          </button>
        </div>
        <div
          class="row"
        >
          <button
            class="number zero"
            type="button"
          >
            0
          </button>
          <button
            class=""
            type="button"
          >
            .
          </button>
          <button
            class="operation"
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
